wordpress-mobile/WordPress-Android
# Add tap and shown tracking to notifications

Fixes #10626

This PR adds tracking for tap and show actions on notifications. We are already tracking that for some notifications (for example for the last notification tapped) but this PR introduces a consistent way so we can check the flow for each notification type. 

I haven't add the `show` tracking to the GCMMessageService because it's static and it would need a lot of refactoring. I'm trying to do that in a separate PR.

To test:
* Make a notification appear in the device (by adding a comment or a post to a followed site - or any other way)
* Check in LogCat that the `notification_shown` event is tracked and it contains the `notification_type` property
* Tap on the notification
* Check in LogCat that the `notification_tapped` event is tracked and it contains the `notification_type` property

PR submission checklist:
